packages/api/src/controllers/taskController.js

```diff
@@ -619,16 +619,30 @@ const taskController = {
       const { user_id } = req.auth;
       const { farm_id } = req.headers;
       const task_id = parseInt(req.params.task_id);
-      const { assignee_user_id, assignee_role_id } = await TaskModel.getTaskAssignee(task_id);
+      const {
+        assignee_user_id,
+        assignee_role_id,
+        wage_at_moment,
+        override_hourly_wage,
+      } = await TaskModel.getTaskAssignee(task_id);
       const { role_id } = await UserFarmModel.getUserRoleId(user_id);
       if (!canCompleteTask(assignee_user_id, assignee_role_id, user_id, role_id)) {
         return res.status(403).send("Not authorized to complete other people's task");
       }
+
+      const wagePatchData = { wage_at_moment: override_hourly_wage ? wage_at_moment : null };
```

Comment:
There's a fair amount of code that is common between `completeHarvestTask` and `completeTask` -- could we abstract it to its own separate function/s so that when we make a change to this logic we can be sure that it applies to both? This way we would prevent bugs like this one where the wage logic was applied for regular tasks but not for a harvest task

Reply:
Thank you for the suggestion, I created two functions. It would be great if you could re-review!